In [1]:
from statistics import mean
Region_A = [10, 15, 12, 8, 14]
Region_B = [18, 20, 16, 22, 25] 
mean_A = mean(Region_A)
mean_B = mean(Region_B)
print(f"mean for region A is:{mean_A} and for region_B is:{mean_B}")


mean for region A is:11.8 and for region_B is:20.2


In [2]:
from statistics import mode
cs = [4,5,2,3,5,4,3,2,4,5]
mode_cs = mode(cs)
print(f"mode of the survey is:{mode_cs}")

mode of the survey is:4


In [3]:
from statistics import median
Department_A = [5000, 6000, 5500, 7000]
Department_B = [4500, 5500, 5800, 6000, 5200]
median_1 = median(Department_A)
median_2 = median(Department_B)
print(f"median salary for department A is:{median_1} and for department B is:{median_2}")

median salary for department A is:5750.0 and for department B is:5500


In [4]:
prices = [25.5, 24.8, 26.1, 25.3, 24.9]
price_range = max(prices) - min(prices)
print(price_range)


1.3000000000000007


In [5]:
from scipy.stats import ttest_ind

group_a = [85, 90, 92, 88, 91]
group_b = [82, 88, 90, 86, 87]

t_statistic, p_value = ttest_ind(group_a, group_b)
print("T-statistic:", t_statistic)
print("P-value:", p_value)


T-statistic: 1.4312528946642733
P-value: 0.19023970239078333


In this case, the t-statistic is 1.4312528946642733, and the p-value is 0.19023970239078333. Since the p-value is greater than the commonly used significance level of 0.05, we fail to reject the null hypothesis. This means that there is not enough evidence to suggest a significant difference in the mean scores between the two groups.

In [6]:
import numpy as np

advertising_expenditure = [10, 15, 12, 8, 14]
sales = [25, 30, 28, 20, 26]

correlation_matrix = np.corrcoef(advertising_expenditure, sales)
correlation_coefficient = correlation_matrix[0,1]

print("Correlation coefficient:", correlation_coefficient)


Correlation coefficient: 0.8757511375750132


In [7]:
from statistics import stdev

heights = [160, 170, 165, 155, 175, 180, 170]
standard_deviation = stdev(heights)
print(standard_deviation)


8.591246929842246


In [8]:
from scipy.stats import linregress

employee_tenure = [2, 3, 5, 4, 6, 2, 4]
job_satisfaction = [7, 8, 6, 9, 5, 7, 6]

slope, intercept, r_value, p_value, std_err = linregress(employee_tenure, job_satisfaction)

print("Slope:", slope)
print("Intercept:", intercept)
print("R-value:", r_value)
print("P-value:", p_value)
print("Standard Error:", std_err)


Slope: -0.4680851063829787
Intercept: 8.595744680851062
R-value: -0.520573376708331
P-value: 0.23095550952297988
Standard Error: 0.34333860006713723


In this case, the slope is 0.4680851063829787, indicating that job satisfaction decreases by 0.4680851063829787 units for each year increase in employee tenure. The intercept is 8.595744680851062, indicating the predicted job satisfaction when the employee tenure is 0. The negative r_value of -0.520573376708331 suggests a moderate negative correlation between employee tenure and job satisfaction. The p_value of 0.23095550952297988 suggests that the correlation is not statistically significant at a common significance level of 0.05. The std_err value of 0.34333860006713723 indicates the accuracy of the predicted job satisfaction values.

In [9]:
from scipy.stats import f_oneway

medication_a = [10, 12, 14, 11, 13]
medication_b = [15, 17, 16, 14, 18]

f_statistic, p_value = f_oneway(medication_a, medication_b)
print("F-statistic:", f_statistic)
print("P-value:", p_value)


F-statistic: 16.0
P-value: 0.003949772803445326


In this case, the F-statistic is 16.0, and the p-value is 0.003949772803445326. Since the p-value is less than the commonly used significance level of 0.05, we can reject the null hypothesis. This means that there is enough evidence to suggest a significant difference in the mean recovery times between the two medications.

In [23]:
import numpy as np

feedback_ratings = [8, 9, 7, 6, 8, 10, 9, 8, 7, 8]
percentile_75 = np.percentile(feedback_ratings, 75)
print(percentile_75)


8.75


In [11]:
from scipy.stats import ttest_1samp

weights = [10.2, 9.8, 10.0, 10.5, 10.3, 10.1]
population_mean = 10

t_statistic, p_value = ttest_1samp(weights, population_mean)
print("T-statistic:", t_statistic)
print("P-value:", p_value)


T-statistic: 1.5126584522688367
P-value: 0.19077595151110102


In this case, the t-statistic is 1.5126584522688367, and the p-value is 0.19077595151110102. Since the p-value is greater than the commonly used significance level of 0.05, we fail to reject the null hypothesis. This means that there is not enough evidence to suggest a significant difference between the mean weight and 10 grams.

In [12]:
from scipy.stats import chi2_contingency

design_a = [100, 120, 110, 90, 95]
design_b = [80, 85, 90, 95, 100]

# Create a contingency table
contingency_table = [design_a, design_b]

chi2_statistic, p_value, _, _ = chi2_contingency(contingency_table)
print("Chi-square statistic:", chi2_statistic)
print("P-value:", p_value)


Chi-square statistic: 6.110658166925435
P-value: 0.19103526314060293


In this case, the chi-square statistic is 6.110658166925435, and the p-value is 0.19103526314060293. Since the p-value is greater than the commonly used significance level of 0.05, we fail to reject the null hypothesis. This means that there is not enough evidence to suggest a significant difference in the click-through rates between the two website designs.

In [13]:
from scipy.stats import t

satisfaction_scores = [7, 9, 6, 8, 10, 7, 8, 9, 7, 8]
confidence_level = 0.95

sample_mean = sum(satisfaction_scores) / len(satisfaction_scores)
sample_standard_deviation = stdev(satisfaction_scores)
sample_size = len(satisfaction_scores)
degrees_of_freedom = sample_size - 1

t_critical = t.ppf((1 + confidence_level) / 2, degrees_of_freedom)
margin_of_error = t_critical * sample_standard_deviation / (sample_size ** 0.5)

lower_bound = sample_mean - margin_of_error
upper_bound = sample_mean + margin_of_error

confidence_interval = (lower_bound, upper_bound)
print("Confidence Interval:", confidence_interval)


Confidence Interval: (7.043561120599888, 8.756438879400113)


In this case, the 95% confidence interval for the population mean satisfaction score is (7.043561120599888, 8.756438879400113). This means that we can be 95% confident that the true population mean satisfaction score falls within this range.

In [15]:
from scipy.stats import linregress

temperature = [20, 22, 23, 19, 21]
performance = [8, 7, 9, 6, 8]

slope, intercept, r_value, p_value, std_err = linregress(temperature, performance)

print("Slope:", slope)
print("Intercept:", intercept)
print("R-value:", r_value)
print("P-value:", p_value)
print("Standard Error:", std_err)


Slope: 0.5
Intercept: -2.9000000000000004
R-value: 0.6933752452815364
P-value: 0.19417134561205843
Standard Error: 0.30000000000000004


In this case, the slope is 0.5, indicating that performance increases by 0.5 units for each degree Celsius increase in temperature. The intercept is -2.9, indicating the predicted performance when the temperature is 0 degrees Celsius. The positive r_value of 0.6933752452815364 suggests a strong positive correlation between temperature and performance. The p_value of 0.19417134561205843 suggests that the correlation is not statistically significant at a common significance level of 0.05. The std_err value of 0.30000000000000004 indicates the accuracy of the predicted performance values.

In [16]:
from scipy.stats import mannwhitneyu

group_a = [4, 3, 5, 2, 4]
group_b = [3, 2, 4, 3, 3]

u_statistic, p_value = mannwhitneyu(group_a, group_b, alternative='two-sided')
print("U-statistic:", u_statistic)
print("P-value:", p_value)


U-statistic: 17.0
P-value: 0.380836480306712


In this case, the U-statistic is 17.0, and the p-value is 0.380836480306712. Since the p-value is greater than the commonly used significance level of 0.05, we fail to reject the null hypothesis. This means that there is not enough evidence to suggest a significant difference in the median preferences between the two groups.

In [17]:
import numpy as np

ages = [25, 30, 35, 40, 45, 50, 55, 60, 65, 70]

q1 = np.percentile(ages, 25)
q3 = np.percentile(ages, 75)

iqr = q3 - q1
print(iqr)


22.5


The interquartile range (IQR) is calculated as the difference between the 75th and 25th percentiles (q3 - q1). It represents the range of the middle 50% of the data.

In [19]:
from scipy.stats import kruskal

algorithm_a = [0.85, 0.80, 0.82, 0.87, 0.83]
algorithm_b = [0.78, 0.82, 0.84, 0.80, 0.79]
algorithm_c = [0.90, 0.88, 0.89, 0.86, 0.87]

h_statistic, p_value = kruskal(algorithm_a, algorithm_b, algorithm_c)
print("H-statistic:", h_statistic)
print("P-value:", p_value)


H-statistic: 9.696947935368053
P-value: 0.007840333026249539


In [20]:
from scipy.stats import linregress

price = [10, 15, 12, 8, 14]
sales = [100, 80, 90, 110, 95]

slope, intercept, r_value, p_value, std_err = linregress(price, sales)

print("Slope:", slope)
print("Intercept:", intercept)
print("R-value:", r_value)
print("P-value:", p_value)
print("Standard Error:", std_err)


Slope: -3.506097560975609
Intercept: 136.3719512195122
R-value: -0.8979991308594848
P-value: 0.03850178234753776
Standard Error: 0.9918303504036147


In [21]:
from statistics import mean, stdev

satisfaction_scores = [7, 8, 9, 6, 8, 7, 9, 7, 8, 7]

mean_score = mean(satisfaction_scores)
standard_deviation = stdev(satisfaction_scores)
sample_size = len(satisfaction_scores)

standard_error = standard_deviation / (sample_size ** 0.5)

print("Standard Error:", standard_error)


Standard Error: 0.30550504633038933


In [24]:
import statsmodels.api as sm
import pandas as pd

advertising_expenditure = [10, 15, 12, 8, 14]
sales = [25, 30, 28, 20, 26]

# Create a DataFrame for the independent variables
data = pd.DataFrame({'Advertising Expenditure': advertising_expenditure})

# Add a constant column for the intercept term
data = sm.add_constant(data)

# Create the model and fit it to the data
model = sm.OLS(sales, data)
results = model.fit()

# Print the summary of the regression analysis
print(results.summary())


                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.767
Model:                            OLS   Adj. R-squared:                  0.689
Method:                 Least Squares   F-statistic:                     9.872
Date:                Sun, 09 Jul 2023   Prob (F-statistic):             0.0516
Time:                        13:14:14   Log-Likelihood:                -9.5288
No. Observations:                   5   AIC:                             23.06
Df Residuals:                       3   BIC:                             22.28
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                              coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------------
const                     

C:\ProgramData\anaconda3\lib\site-packages\statsmodels\stats\stattools.py:74: ValueWarning: omni_normtest is not valid with less than 8 observations; 5 samples were given.
  warn("omni_normtest is not valid with less than 8 observations; %i "
